File: go/euler33.go
//
// projecteuler.net
//
// problem 32
//

package main

import (
	"fmt"
)

func gcd(a, b int) int {
	if b == 0 {
		return a
	}
	return gcd(b, a%b)
}

func main() {
	dp := 1
	np := 1
	for i := 1; i < 10; i++ {
		for d := 1; d < i; d++ {
			for n := 1; n < d; n++ {
				if (n*10+i)*d == n*(i*10+d) {
					dp = d * dp
					np = n * np
				}
			}
		}
	}
	dp /= gcd(dp, np)
	answer := dp
	fmt.Printf("%v\n", answer)
}

File: python/euler33.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
# projecteuler.net
#
# problem 33
#

from math import gcd

dp = 1
np = 1

for i in range(11):
    for d in range(1, i):
        for n in range(1, d):
            if (n*10+i)*d == n*(i*10+d):
                dp = d * dp
                np = n * np

answer = int(dp / gcd(dp, np))

print(answer)
